color_picker/src/main/ets/components/MainPage/colorDialog.ets

```diff
@@ -0,0 +1,80 @@
+/*
+* Copyright (c) 2022 Application Library Engineering Group.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+@CustomDialog
+struct colorDialog {
+  @State h: number = 0
+  @State s: number = 100
+  @State l: number = 50
+  @State col: string = '#ff0000'
+  controller: CustomDialogController;
+  action: (number) => void;
+
+  rgbToHex(r, g, b) {
+    return "#" + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
+  }
+
+  HSLToRGB(h, s, l) {
+    s /= 100;
+    l /= 100;
+    const k = n => (n + h / 30) % 12;
+    const a = s * Math.min(l, 1 - l);
+    const f = n =>
+    l - a * Math.max(-1, Math.min(k(n) - 3, Math.min(9 - k(n), 1)));
+    return [Math.round(255 * f(0)), Math.round(255 * f(8)), Math.round(255 * f(4))];
+  }
+
+  build() {
+    Column() {
+      Flex({ justifyContent: FlexAlign.SpaceEvenly }) {
+        Button('OK')
+          .width(60)
```

Comment:
Use 'fp' for font/text size or 'vp' for all other dimensions.

Reply:
Made concerned changes